### ISSUE TITLE
2 minute stall on "Getting code actions from "Go"" every 20th save even though actions disabled

### ISSUE THREAD

**Issue by tonimelisma** [golang/vscode-go#1028]
### What version of Go, VS Code & VS Code Go extension are you using?
- Run `go version` to get version of Go
	- go 1.15.6
- Run `gopls -v version` to get version of Gopls if you are using [the language server](https://github.com/golang/vscode-go/blob/master/docs/gopls.md).
	- v0.6.0-pre1 and v0.5.5
- Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders
	- Version: 1.53.0-insider
Commit: 76436a4d43566cd313629799e2ec51b4d00caa64
Date: 2020-12-11T17:46:02.603Z
Electron: 11.0.3
Chrome: 87.0.4280.67
Node.js: 12.18.3
V8: 8.7.220.25-electron.0
OS: Darwin x64 20.1.0
- Check your installed extensions to get the version of the VS Code Go extension
	- v0.19.1 as well as go-nightly v2020.12.814
- Run `go env` to get the go development environment details
	- GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/toni/.cache/go-build"
GOENV="/home/toni/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/toni/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/toni/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/toni/sdk/go1.15.6"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/toni/sdk/go1.15.6/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/toni/go/src/github.com/davidbyttow/govips/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build840047076=/tmp/go-build -gno-record-gcc-switches"

### Share the Go related settings you have added/edited

Run `Preferences: Open Settings (JSON)` command to open your settings.json file.
Share all the settings with the `go.` or `["go"]` or `gopls` prefixes.
"go.formatTool": "goimports",
    "go.useLanguageServer": true,
   "go.coverageDecorator": {
        "type": "gutter"
    },
    "go.coverOnSingleTest": true,
    "go.coverOnSingleTestFile": true,
  "go.coverageOptions": "showUncoveredCodeOnly",
   "editor.codeActionsOnSave": null

### Describe the bug
Every now and then, for months now with various Go extensions and Go nightly extensions, when I save a Go file, VS code stalls for a couple of minutes saying "Getting code actions from Go"

The stall is about two minutes. As you can see from the screenshot, editor.CodeActionsOnSave is null.

### Steps to reproduce the behavior:
1. Edit go file
2. Save go file
3. Every 20th time VS code hangs for a few minutes, saving the file

### Screenshots or recordings
If applicable, add screenshots or recordings to help explain your problem.

<img width="933" alt="Screenshot 2020-12-12 at 21 48 47" src="https://user-images.githubusercontent.com/59022391/101993593-88b8c880-3cc4-11eb-996f-054eb7346072.png">

**Comment by stamblerre** [comment]
If you want to disable code actions, I think the right syntax is `"editor.codeActionsOnSave": {}`. If you'd like to investigate the underlying issue here, please share a `gopls` log: https://github.com/golang/tools/blob/master/gopls/doc/troubleshooting.md#capturing-logs.

**Comment by tonimelisma** [comment]
Hey @hyangah here's a trace.

[vscode_hangup.txt](https://github.com/golang/vscode-go/files/5721599/vscode_hangup.txt)
<img width="1017" alt="Screenshot 2020-12-21 at 0 02 58" src="https://user-images.githubusercontent.com/59022391/102725910-0c0c9680-4323-11eb-9921-fd1e0968b2b8.png">

**Comment by stamblerre** [comment]
It looks like your log may be incomplete--I would expect it to start with an `initialize` request. Do you mind sharing a complete log starting from the initialization?

Also, I notice that there are 2 extensions listed in the pop-up here--does disabling the "GitHub Pull Requests and Issues" extension fix the problem?

**Comment by tonimelisma** [comment]
@stamblerre did you read the attached text file as well? It covers the whole period of the hangup, not all they way from the start of vs code. Isn't that enough?

**Comment by stamblerre** [comment]
Yes, I did read that file, but `gopls` logs useful debugging information at start-up, which is why it would be best to have a log starting at initialization. If that's not possible, you can run `gopls -rpc.trace -v check path/to/file.go` in your VS Code workspace root.

**Comment by tonimelisma** [comment]
@stamblerre here's one all the way from the beginning
[Untitled.txt](https://github.com/golang/vscode-go/files/5731579/Untitled.txt)

**Comment by tonimelisma** [comment]
On further usage, right now, when I e.g. run the tests, the Go extension starts compiling the program. If I cancel the tests, it doesn't stop the compilation, it keeps running on the remote server.

**Comment by tonimelisma** [comment]
Hey @stamblerre any updates on the topic? I suppose we are still not "waiting for info"? This makes VS code impossible to use for me

**Comment by stamblerre** [comment]
Apologies, I must've missed your response. Does setting `"go.useLanguageServer": false` resolve the problem temporarily while we investigate?

**Comment by stamblerre** [comment]
> On further usage, right now, when I e.g. run the tests, the Go extension starts compiling the program. If I cancel the tests, it doesn't stop the compilation, it keeps running on the remote server. Now every time I save a Go file it stalls for 2 minutes as it's waiting for the compilation to end.
> 
> Edit: actually, whenever I save a file, it starts compiling the whole program. If I save numerous time one after another, for each save, it starts the whole process of compiling the program. It's a cgo program.

Ah, I see it's also related to running the tests. Does it work if you don't run the tests, or are the saves still slow then?

**Comment by tonimelisma** [comment]
It's unrelated to testing, although "Build on save" is set to package, the default value

**Comment by stamblerre** [comment]
That setting doesn't actually apply when the language server is enabled. To follow up on some of my earlier questions:

- Were you able to disable code actions by setting `"editor.codeActionsOnSave": {}`?
- Does disabling the "GitHub Pull Requests and Issues" have any effect?
- Does setting "go.useLanguageServer": false resolve the problem temporarily while we investigate?

**Comment by tonimelisma** [comment]
Hey @stamblerre still seeing this issue.
- Disabling Github Pull requests doesn't have any effect
- Setting language server false does indeed help
- "editor.codeActionsOnSave": {} indeed does remove the stalling yet on saving it does run gofmt (or similar) on my file

**Comment by stamblerre** [comment]
> "editor.codeActionsOnSave": {} indeed does remove the stalling yet on saving it does run gofmt (or similar) on my file

You can set `"editor.formatOnSave": false` to disable formatting.

**Comment by tonimelisma** [comment]
That's fine. The big issue still is that it tries to build my Go project each time I save the file. Being a cgo project, that takes several minutes.

**Comment by stamblerre** [comment]
Is this still happening with `"go.useLanguageServer": false`? If so, then you can try setting `"go.buildOnSave": "off"`.

**Comment by tonimelisma** [comment]
Hey @stamblerre please see my previous comment 6 hours ago, in which I explained that it is indeed not happening when I set go.useLanguageServer false.
